### PR TITLE
IA-2565: Org units search change source should reset parent org unit

### PR DESCRIPTION
**Pull request by beygorghor**
If source field isn't actively cleared before switching source, then parent selection doesn't take into account the source switch
In Org unit search, if source field isn't actively cleared before switching source, then parent selection doesn't take into account the source switch (see attached video).

https://github.com/BLSQ/iaso/assets/12494624/e04a1149-0cf2-4cbe-aec2-a55d455b4ad4

Related JIRA tickets : IA-2565

## Self proofreading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- fix typings issues
- delete levels in search if source is changed

## How to test

Make a string search leading to few results on a specific source. (take a print screen to compare results)
Change the source and choose a  parent org unit. and keep the same string search and search.
List should be empty or different.
Switch back to first source, keep search string and search again, results should be the same as the print screen.

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/228a1d4b-4d64-44c3-ad40-3a592c5488b6

## Notes

Cypress test are not implemented here, we should keep in mind those race conditions and test it !

### CODE REVIEW

File: hat/assets/js/apps/Iaso/components/filters/DatesRange.d.ts
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/filters/DatesRange.d.ts:1]: How was this generated?

File: hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitsFilters.tsx [hunk 1]
```diff
@@ -117,20 +117,22 @@ export const OrgUnitFilters: FunctionComponent<Props> = ({
...
     } = useGetValidationStatus(true);
     const handleChange = (key, value) => {
+        const newFilters: Record<string, unknown> = {
+            ...filters,
+        };
         if (key === 'version') {
             setSourceVersionId(parseInt(value, 10));
         }
         if (key === 'source') {
+            console.log('Reset');
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitsFilters.tsx]: 🚮

File: hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitsFilters.tsx [hunk 2]
```diff
@@ -258,7 +260,7 @@ export const OrgUnitFilters: FunctionComponent<Props> = ({
                     label={MESSAGES.search}
                     blockForbiddenChars
                     onEnterPressed={onSearch}
-                    onErrorChange={setTextSearchError}
+                    onErrorChange={(hasError) => setTextSearchError(hasError)}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitsFilters.tsx:263]: This is just to make the TS compiler happy isn't it?😁

**Review by quang-le — CHANGES_REQUESTED** [review]
I noticed that in the url, when we change source, the `orgUnitParentId` is not cleared
https://github.com/BLSQ/iaso/assets/38907762/30b7fd61-f9a7-449f-9607-ff379b537ccb
It doesn't break anything, but since we're cleaning up...

File: hat/assets/js/apps/Iaso/components/filters/DatesRange.d.ts
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/components/filters/DatesRange.d.ts:1]: by hand

File: hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitsFilters.tsx (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/orgUnits/components/OrgUnitsFilters.tsx:263]: yep

**Review by quang-le — APPROVED** [review]

**Pull request merged**